crates/nu-command/src/maybe_print_errors.rs

```diff
@@ -1,17 +0,0 @@
-use nu_engine::EvaluationContext;
-use nu_source::Text;
-
-pub fn maybe_print_errors(context: &EvaluationContext, source: Text) -> bool {
```

Comment:
This feels like something that would make a good callback to pass into engine, rather than living directly in engine.

It would help keep the engine a bit more "pure" in that it doesn't have to know anything about the context it's running in, it can just call into a function that's passed in that knows how to into interact with the host and can print out the errors.

Reply:
I feel like this function should live in nu-engine, as printing errors is core functionality necessary to drive nu.

I agree that currently it is not "pure" - it does not use Host::stdout / Host::stderr. Would you still like to see this function outside of nu-engine, even if it would use Host::stdout / Host::stderr?

Reply:
#3112

Reply:
Update: I have rebased this PR onto #3112 and moved the print functions into their own module at: `nu-engine/src/print.rs`. I hope that leads to some better separation and modularization :)